src/OSSupport/Semaphore.cpp

```diff
@@ -6,32 +6,22 @@
 
 
 cSemaphore::cSemaphore( unsigned int a_MaxCount, unsigned int a_InitialCount /* = 0 */)
-#ifndef _WIN32
-	: m_bNamed( false)
-#endif
 {
 #ifndef _WIN32
 	(void)a_MaxCount;
 	m_Handle = new sem_t;
-	if (sem_init( (sem_t*)m_Handle, 0, 0))
+	AString Name;
+	Printf(Name, "cSemaphore%p", this);
```

Comment:
This is broken. Semaphore names are system global, so this could cause issues running multiple copies of MCServer at once.